### ISSUE TITLE
Error message does not have SQL with newest DBI even if ShowStatement was enabled

### ISSUE THREAD

**Issue by kazeburo** [DBD-SQLite/DBD-SQLite#18]
```
use 5.10.0;
use DBI;
use DBD::SQLite;
use DBD::mysql;
use Test::More;

say "DBI:$DBI::VERSION";
say "DBD::SQLite:$DBD::SQLite::VERSION";
say "DBD::mysql:$DBD::mysql::VERSION";

my $dbh;
if ( $ARGV[0] eq 'mysql' ) {
    $dbh = DBI->connect('dbi:mysql:test', '', '',{
        RaiseError => 1,
        PrintError => 0,
        ShowErrorStatement => 1,
        AutoInactiveDestroy => 1,
    });
    $dbh->do(q{DROP TABLE foo});
}
else {
    $dbh = DBI->connect('dbi:SQLite::memory:', '', '',{
        RaiseError => 1,
        PrintError => 0,
        ShowErrorStatement => 1,
        AutoInactiveDestroy => 1,
    });
}

$dbh->do(q{CREATE TABLE foo (
    id INTEGER NOT NULL PRIMARY KEY,
    e VARCHAR(10)
)});

eval {
    my $sth = $dbh->prepare(q{INSERT INTO foo (e2) VALUES (?)});
    $sth->execute(1);
};

like $@, qr/for Statement/;

done_testing;
```

DBI@1.634

```
$ perl sqlite.pl 
DBI:1.634
DBD::SQLite:1.50
DBD::mysql:4.027
ok 1
1..1
$ perl sqlite.pl  mysql
DBI:1.634
DBD::SQLite:1.50
DBD::mysql:4.027
ok 1
1..1
```

DBI@1.636

```
$ perl sqlite.pl  mysql
DBI:1.636
DBD::SQLite:1.50
DBD::mysql:4.027
ok 1
1..1
$ perl sqlite.pl       
DBI:1.636
DBD::SQLite:1.50
DBD::mysql:4.027
not ok 1
#   Failed test at sqlite.pl line 40.
#                   'DBD::SQLite::db prepare failed: table foo has no column named e2 at sqlite.pl line 36.
# '
#     doesn't match '(?^:for Statement)'
1..1
# Looks like you failed 1 test of 1.
```

**Comment by charsbar** [comment]
This seems to be an issue on DBI: https://github.com/perl5-dbi/dbi/pull/34

**Comment by timbunce** [comment]
Interesting. Could you attach a trace of the failing case? E.g. set env var `DBI_TRACE=6=dbi.log`

**Comment by charsbar** [comment]
[dbi_1636_sqlite_1504_log.txt](https://github.com/DBD-SQLite/DBD-SQLite/files/337006/dbi_1636_sqlite_1504_log.txt)

**Comment by charsbar** [comment]
@timbunce, hope ^^ helps.

**Comment by charsbar** [comment]
This issue was fixed by DBI 1.637. Thanks.
